### DISCUSSION TITLE
Input does not output the value attribute if no name was provided

### DISCUSSION THREAD

**Discussion by raveren** [spatie/laravel-html#233] [Q&A]
```php
{{ html()->text(value: 'my value') }}
```

The generated html has no `value` attribute due to this line checking for `$name` being true-ish: https://github.com/spatie/laravel-html/blob/main/src/Html.php#L259 - for no apparent reason.

I explicitly provide a value, I expect the value to be displayed, is there anyone who can explain why would I not?

**Comment by raveren** [comment]
To those who just want a workaround, you can use:

```
{{ html()->text()->value('my value') }}
```

**Comment by freekmurze** [comment]
This seems like a bug, and we'd welcome a PR.

**Reply by raveren** [reply]
https://github.com/spatie/laravel-html/pull/235
